waspc/data/Generator/templates/server/src/auth/providers/email/utils.ts

```diff
@@ -2,80 +2,68 @@
-  try {
-    await prisma.{= userEntityLower =}.update({
-      where: { id: userId },
-      data: { isEmailVerified: true },
-    })
-  } catch (e) {
-    rethrowPossiblePrismaError(e);
-  }
+type AuthWithId = {
+  id: {= authEntityUpper =}Id,
 }
 
-export async function updateUserPassword(userId: {= userEntityUpper =}Id, password: string): Promise<void> {
-  try {
-    await prisma.{= userEntityLower =}.update({
-      where: { id: userId },
-      data: { password },
-    })
-  } catch (e) {
-    rethrowPossiblePrismaError(e);
-  }
-}
-
-export async function createEmailVerificationLink(user: {= userEntityUpper =}, clientRoute: string): Promise<string> {
-  const token = await createEmailVerificationToken(user);
+export async function createEmailVerificationLink(auth: AuthWithId, clientRoute: string) {
+  const token = await createEmailVerificationToken(auth);
   return `${waspServerConfig.frontendUrl}${clientRoute}?token=${token}`;
 }
 
-export async function createPasswordResetLink(user: {= userEntityUpper =}, clientRoute: string): Promise<string> {
-  const token = await createPasswordResetToken(user);
+export async function createPasswordResetLink(auth: AuthWithId, clientRoute: string)  {
+  const token = await createPasswordResetToken(auth);
   return `${waspServerConfig.frontendUrl}${clientRoute}?token=${token}`;
 }
 
-async function createEmailVerificationToken(user: {= userEntityUpper =}): Promise<string> {
-  return sign(user.id, { expiresIn: '30m' });
+async function createEmailVerificationToken(auth: AuthWithId): Promise<string> {
+  return sign(auth.id, { expiresIn: '30m' });
 }
 
-async function createPasswordResetToken(user: {= userEntityUpper =}): Promise<string> {
-  return sign(user.id, { expiresIn: '30m' });
+async function createPasswordResetToken(auth: AuthWithId): Promise<string> {
+  return sign(auth.id, { expiresIn: '30m' });
 }
 
 export async function sendPasswordResetEmail(
   email: string,
   content: Email,
-): Promise<void> {
+) {
   return sendEmailAndLogTimestamp(email, content, 'passwordResetSentAt');
 }
 
 export async function sendEmailVerificationEmail(
   email: string,
   content: Email,
-): Promise<void> {
+) {
   return sendEmailAndLogTimestamp(email, content, 'emailVerificationSentAt');
 }
 
 async function sendEmailAndLogTimestamp(
   email: string,
   content: Email,
   field: 'emailVerificationSentAt' | 'passwordResetSentAt',
-): Promise<void> {
+) {
   // Set the timestamp first, and then send the email
   // so the user can't send multiple requests while
   // the email is being sent.
   try {
-    await prisma.{= userEntityLower =}.update({
-      where: { email },
-      data: { [field]: new Date() },
-    })
+    const authIdentity = await findAuthIdentity("email", email);
+    const providerData = deserializeProviderData<'email'>(authIdentity.providerData);
+    await updateAuthIdentityProviderData<'email'>(authIdentity.authId, providerData, {
+        [field]: new Date()
```

Comment:
Since both Date's are coming from the same server, the TZ will match and then when we compute the time diff, it will be good. I'm not sure if there are any more advanced scenarios I missed?

Reply:
I am not sure we should be relying on them happening on the same server. What if I am migrating from one server to another, and migration is done quite quickly?

Reply:
I've updated the code use the UTC date 👍